contract/contracts/reverse_name_resolver/reverse_name_resolver.ral

```diff
@@ -0,0 +1,39 @@
+Contract ReverseNameResolver() extends Constants(), Events() {
+
+    mapping[Address, ByteVec] addressNames
+
+    pub fn getNameByAddress(address: Address) -> ByteVec {
+        return addressNames[address]
+    }
+
+    pub fn containsNameByAddress(address: Address) -> Bool {
+        return addressNames.contains!(address)
+    }
+
+    @using(preapprovedAssets = true, checkExternalCaller = false)
+    pub fn setAddressName(name: ByteVec) -> () {
```

Comment:
It seems any address could set an arbitrary name for itself. For example, one could register its address to `test.alph` even if `test.alph` is not attached to the address.

Is this expected behavior?

Reply:
This is expected yet. They can set whatever address they like but it will only be considered "Linked" once the address has created a mapping in the reverse direction in one of the 4 `ReverseNameResolver` contracts. We can't confirm permissions here because the target can be on a different Group.